### PR TITLE
#253: Add scroll feature

### LINKED ISSUE #253: Add option to scroll to the top
## I'm submitting a
- [ ] Bug report
- [x] Feature request

## Expected behavior
Add an option to scroll to the top of the page like in the [EvalAI-cli docs](https://evalai-cli.cloudcv.org/) page.

### PR DESCRIPTION
**Pull request by drepram**
Fixes #253

#### Changes proposed in this pull request:

- Added button to scroll back up

- Link to live demo: http://pr-263-evalai.surge.sh 

- presence of footer button in EvalAI homepage

### CODE REVIEW

**Comment by Sanji515** [comment]
@drepram Can we animate while we scroll to the top like [here](https://evalai-cli.cloudcv.org/)?

**Comment by Sanji515** [comment]
@drepram We already have a scroll up button [here](https://github.com/Cloud-CV/EvalAI-ngx/blob/master/src/app/components/publiclists/challengelist/challengelist.component.html#L17). Can we use this one? With the same style and same smooth behaviour.

File: src/app/app.component.html [hunk 1]
```diff
@@ -6,3 +6,6 @@
 <app-modal *ngIf="modalParams['isModalVisible']" [params]="modalParams"></app-modal>
 <app-editphasemodal *ngIf="editPhaseModalParams['isEditPhaseModalVisible']" [params]="editPhaseModalParams"></app-editphasemodal>
 <app-terms-and-conditions-modal *ngIf="termsAndConditionsModalParams['isTermsAndConditionsModalVisible']" [params]="termsAndConditionsModalParams"></app-terms-and-conditions-modal>
+<div id = "scroll-up" *ngIf="isScrollbtnVisible" (click)="scrollUp()" >
```
Comment (Sanji515) [inline src/app/app.component.html]: Please remove the space before and after `=`.

**Comment by Sanji515** [comment]
@drepram I don't see the button on the homepage and at other places also, can you please check?

**Comment by drepram** [comment]
Screen Shot 2020-01-10 at 18 06 54

On careful consideration, I think the yellow button style looks way better than the current one, shall I replace the one on the allchallenges.ts file instead?

File: src/app/app.component.ts [hunk 2]
```diff
@@ -143,4 +146,10 @@ export class AppComponent implements OnInit, OnDestroy {
       this.globalServiceSubscriptionScrollTop.unsubscribe();
     }
   }
+
+  topFunction() {
+    document.body.scrollIntoView({behavior: 'smooth'});
+    // document.body.scrollTop = 0;
+    // document.documentElement.scrollTop = 0;
```
Comment (Sanji515) [inline src/app/app.component.ts]: @drepram Can you please remove these comments here?

File: src/app/app.component.html [hunk 3]
```diff
@@ -6,3 +6,4 @@
 <app-modal *ngIf="modalParams['isModalVisible']" [params]="modalParams"></app-modal>
 <app-editphasemodal *ngIf="editPhaseModalParams['isEditPhaseModalVisible']" [params]="editPhaseModalParams"></app-editphasemodal>
 <app-terms-and-conditions-modal *ngIf="termsAndConditionsModalParams['isTermsAndConditionsModalVisible']" [params]="termsAndConditionsModalParams"></app-terms-and-conditions-modal>
+<i (click)="topFunction()" class="fa fa-arrow-up" id="up-arrow"></i>
```
Comment (Sanji515) [inline src/app/app.component.html]: please change `topFunction()` --> `scrollUp()`.

File: src/app/app.component.ts [hunk 4]
```diff
@@ -18,6 +18,7 @@ import { Title } from '@angular/platform-browser';
 export class AppComponent implements OnInit, OnDestroy {
   private scrolledState = false;
   isLoading = false;
+  showScrollButton = false;
```
Comment (Sanji515) [inline src/app/app.component.ts]: I think we are not using it, can you please remove this?

**Review by Sanji515 — COMMENTED** [review]
Minor changes, Rest LGTM!

**Review by Sanji515 — APPROVED** [review]
Thanks @drepram for adding scroll feature. This can be merged :tada:

**Review by RishabhJain2018 — APPROVED** [review]

**Comment by drepram** [comment]
Thank you to you all, it has been my pleasure.

**Pull request merged**